test/jdk/java/net/httpclient/lib/jdk/httpclient/test/lib/common/HttpServerAdapters.java

```diff
@@ -782,6 +782,11 @@ static void enableLogging() {
         public abstract Version getVersion();
         public abstract void setRequestApprover(final Predicate<String> approver);
 
+        @Override
+        public void close() throws Exception {
+            stop();
+        }
+
```

Comment:
The Http2TestServer is itself closeable, so you should override this method in Http2TestServerImpl below to call `impl.close()`.

Reply:
1. If the impl is AutoCloseable then close() should call close().
2. I would actually prefer to make all implementations AutoCloseable, but that's not the point of this PR.

Reply:
Implemented changes as requested.